homeassistant/components/media_player/bluesound.py

```diff
@@ -0,0 +1,812 @@
+            self._status = xmltodict.parse(result)['status'].copy()
+            self.schedule_update_ha_state()
+
+        except (asyncio.TimeoutError, ClientError):
+            self._is_online = False
+            self._last_status_update = None
+            self._status = None
+            self.schedule_update_ha_state()
+            _LOGGER.info("Client connection error, marking %s as offline",
+                         self._name)
+            raise
+        finally:
+            if response is not None:
+                yield from response.release()
+
+    @asyncio.coroutine
+    @Throttle(SYNC_STATUS_INTERVAL)
+    def async_update_sync_status(self, on_updated_cb=None,
+                                 raise_timeout=False):
+        """Update sync status."""
+        self._internal_update_sync_status(on_updated_cb, raise_timeout=False)
+
+    @asyncio.coroutine
+    @Throttle(UPDATE_CAPTURE_INTERVAL)
+    def async_update_captures(self):
+        """Update Capture cources."""
+        resp = yield from self.send_bluesound_command(
+            'RadioBrowse?service=Capture')
+        if not resp:
+            return
+        self._capture_items = []
+
+        def _create_capture_item(item):
+            self._capture_items.append({
+                'title': item.get('@text', ''),
+                'name': item.get('@text', ''),
+                'type': item.get('@serviceType', 'Capture'),
+                'image': item.get('@image', ''),
+                'url': item.get('@URL', '')
+            })
+
+        if 'radiotime' in resp and 'item' in resp['radiotime']:
+            if isinstance(resp['radiotime']['item'], list):
+                for item in resp['radiotime']['item']:
+                    _create_capture_item(item)
+            else:
+                _create_capture_item(resp['radiotime']['item'])
+
+        return self._capture_items
+
+    @asyncio.coroutine
+    @Throttle(UPDATE_PRESETS_INTERVAL)
+    def async_update_presets(self):
+        """Update Presets."""
+        resp = yield from self.send_bluesound_command('Presets')
+        if not resp:
+            return
+        self._preset_items = []
+
+        def _create_preset_item(item):
+            self._preset_items.append({
+                'title': item.get('@name', ''),
+                'name': item.get('@name', ''),
+                'type': 'preset',
+                'image': item.get('@image', ''),
+                'is_raw_url': True,
+                'url2': item.get('@url', ''),
+                'url': 'Preset?id=' + item.get('@id', '')
+            })
+
+        if 'presets' in resp and 'preset' in resp['presets']:
+            if isinstance(resp['presets']['preset'], list):
+                for item in resp['presets']['preset']:
+                    _create_preset_item(item)
+            else:
+                _create_preset_item(resp['presets']['preset'])
+
+        return self._preset_items
+
+    @asyncio.coroutine
+    @Throttle(UPDATE_SERVICES_INTERVAL)
```

Comment:
That's interesting, I did not expect that but I just tested it and it seems like you're right. Okay, you can leave them in.